### PR TITLE
`d/azurerm_virtual_machine_scale_set`: Add scale set instances details

### PR DESCRIPTION
**Pull request by mouellet**
### Summary

Outputs the list of virtual machine instance currently in the scale set w/ details including private/public ip adresses, computer name, etc. Useful when using auto scaling...

Supersede #14031

### Acceptance tests
```
make acctests SERVICE='compute' TESTARGS='-run=TestAccDataSourceVirtualMachineScaleSet_basicLinux' TESTTIMEOUT='10m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/compute -run=TestAccDataSourceVirtualMachineScaleSet_basicLinux -timeout 10m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceVirtualMachineScaleSet_basicLinux
=== PAUSE TestAccDataSourceVirtualMachineScaleSet_basicLinux
=== CONT  TestAccDataSourceVirtualMachineScaleSet_basicLinux
--- PASS: TestAccDataSourceVirtualMachineScaleSet_basicLinux (245.95s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/compute	247.273s
```

### CODE REVIEW

File: internal/services/compute/virtual_machine_scale_set_data_source.go [hunk 1]
```diff
@@ -86,6 +162,12 @@ func dataSourceVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, meta interf
 	}
 	d.SetId(id.ID())
 
+	if location := resp.Location; location != nil {
+		if err := d.Set("location", azure.NormalizeLocation(*location)); err != nil {
+			return fmt.Errorf("setting `location`: %+v", err)
+		}
+	}
```
Comment (tombuildsstuff) [inline internal/services/compute/virtual_machine_scale_set_data_source.go:169]:
```suggestion
	d.Set("location", location.Normalize(resp.Location))
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @mouellet

Apologies for the delayed review on this one - we've been trying to work out how to proceed here. Having chatted about this internally we think this is fine and have left some comments inline, if you can fix those up then we should be able to take another look and get this merged 👍 

Thanks!

File: internal/services/compute/virtual_machine_scale_set_data_source.go [hunk 2]
```diff
@@ -102,5 +184,143 @@ func dataSourceVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, meta interf
...
+		return fmt.Errorf("listing VM Instances for Virtual Machine Scale Set %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	for result.NotDone() {
+		instance := result.Value()
+		if err := result.NextWithContext(ctx); err != nil {
+			return fmt.Errorf("listing next page VM Instances for Virtual Machine Scale Set %q (Resource Group %q): %+v", name, resGroup, err)
+		}
+
+		nics, err := networkInterfacesClient.ListVirtualMachineScaleSetVMNetworkInterfacesComplete(ctx, resGroup, name, *instance.InstanceID)
```
[the remark this answers is not in the document]
Comment (mouellet) [reply internal/services/compute/virtual_machine_scale_set_data_source.go]: I've added a nil check here! However, I couldn't find any underlying properties (even when adding the expand w/ instanceView parameter) that hold this attribute in the client or documentation (see: https://docs.microsoft.com/en-us/rest/api/compute/virtual-machine-scale-set-vms/list).

**Comment by mouellet** [comment]
Hi @tombuildsstuff, Thanks for the review! I've rebased and pushed the changes you suggested.. Let me know if there's anything else I can do!

**Review by katbyte — APPROVED** [review]
Thanks @mouellet - LGTM ♻️

**Pull request merged**